packages/expo-cli/src/prompt.ts

```diff
@@ -11,8 +11,21 @@ export default function prompt(
   questions: CliQuestions,
   { nonInteractiveHelp }: { nonInteractiveHelp?: string } = {}
 ) {
-  const nQuestions = Array.isArray(questions) ? questions.length : 1;
-  if (program.nonInteractive && nQuestions !== 0) {
+  questions = Array.isArray(questions) ? questions : [questions];
+  const nAllQuestions = questions.length;
+  if (program.nonInteractive) {
+    const nQuestionsToAsk = questions.filter(question => {
+      if (!('when' in question)) {
+        return true;
+      } else if (typeof question.when === 'function') {
+        return question.when({});
```

Comment:
It's a bit confusing for me when i read this and i need to look up `inquirer` docs to make sense of what kind of object `when` should be. It'd be helpful if you had some comments explaining that `when` should be :
- boolean 
- function: and if it is a function,  it expects to take the user's answers object, which is empty in the case of `non-interactive`